### ISSUE TITLE
Fixes for LT01 not applied...

### ISSUE THREAD

**Issue by matepek** [sqlfluff/sqlfluff#5453]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```
WARNING    Fixes for LT01 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning. 
```

### Expected Behaviour

formats the code

### Observed Behaviour

error

### How to reproduce

Small code: 
```
with y AS 
(SELECT *
FROM x
WHERE true
  AND function_name(CAST(ARRAY['element1'] AS ARRAY<STRING>))
)
select * from y
```

### Dialect

bigquery

### Version

sqlfluff, version 2.3.5

### Configuration

```
[sqlfluff]
dialect = bigquery
templater = dbt
sql_file_exts = .sql

# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/configuration.html#enabling-and-disabling-rules
# See https://docs.sqlfluff.com/en/stable/configuration.html#ignoring-errors-files
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
exclude_rules = structure.column_order, capitalisation.identifiers

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
max_line_length = 0

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = 4

[sqlfluff:templater:dbt]
project_dir = ./
profiles_dir = ./
profile = bq
target = dev

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 4
indented_joins = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
